### ISSUE TITLE
Some unit tests of python client are failing on Windows

### ISSUE THREAD

**Issue by JohanKJSchreurs** [Open-EO/openeo-python-client#393]
### Some unit tests are failing on Windows

See pytest report below:

```
=========================================================================================== short test summary info ============================================================================================
SKIPPED [2] tests\udf\test_xarraydatacube.py:255: _save_DataArray_to_NetCDF introduces band names if they don't exist, which fails equality test
SKIPPED [2] tests\udf\test_xarraydatacube.py:265: _save_DataArray_to_NetCDF introduces band names if they don't exist, which fails equality test
SKIPPED [2] tests\udf\test_xarraydatacube.py:275: _save_DataArray_to_NetCDF introduces band dim if it doesn't exist, which fails equality test
FAILED tests/test_config.py::test_get_config_verbose[print-False-True] - assert None
FAILED tests/test_config.py::test_get_config_verbose[print-True-True] - assert None
FAILED tests/test_config.py::test_get_config_verbose[auto-False-False] - assert None
FAILED tests/test_config.py::test_get_config_verbose[auto-True-True] - assert None
FAILED tests/test_config.py::test_get_config_verbose[off-False-False] - assert None
FAILED tests/test_config.py::test_get_config_verbose[off-True-False] - assert None
FAILED tests/rest/test_userfile.py::TestUserFile::test_upload - AssertionError: assert b'hello world\r\n' == b'hello world\n'
FAILED tests/rest/test_userfile.py::TestUserFile::test_connection_upload[None-to-upload.txt] - AssertionError: assert b'hello world\r\n' == b'hello world\n'
FAILED tests/rest/test_userfile.py::TestUserFile::test_connection_upload[foo.txt-foo.txt] - AssertionError: assert b'hello world\r\n' == b'hello world\n'
FAILED tests/rest/test_userfile.py::TestUserFile::test_connection_upload[data/foo.txt-data/foo.txt] - AssertionError: assert b'hello world\r\n' == b'hello world\n'
========================================================================== 10 failed, 1203 passed, 6 skipped, 146 warnings in 39.74s ===========================================================================

```` 

### Possible causes

At least for a few of the tests, it looks like the cause is that the assert compares strings with newlines but the EOL characters are different on Windows,  "\r\n" instead of "\n"

The others seem to be related to paths to the config files, which will be a bit different on Windows. Could be an incorrect escape sequence in the test perhaps.

**Comment by soxofaan** [comment]
a quickfix for the failing TestUserFile tests is probably to test with binary files instead of these "text" ones 
https://github.com/Open-EO/openeo-python-client/blob/3ab4528e4c1570f557d7e3a9fe29558a89bc58e4/tests/rest/test_userfile.py#L71

**Comment by soxofaan** [comment]
fixed through PRs #394 and #396
